glib/src/gstring.rs

```diff
@@ -62,13 +61,15 @@ impl GString {
     /// The underlying string must not be mutated, in particular in terms of
     /// length, underneath the `GString` instance for the duration of the borrow.
     unsafe fn new_borrowed(ptr: *const c_char) -> Borrowed<Self> {
-        assert!(!ptr.is_null());
         Borrowed::new(GString(Inner::Foreign(ptr as *mut _, libc::strlen(ptr))))
     }
 
     pub fn as_str(&self) -> &str {
         let cstr = match self {
             GString(Inner::Foreign(ptr, length)) => unsafe {
+                if ptr.is_null() || length == &0 {
```

Comment:
Why `&0` instead of `0`? :)

Reply:
Because length is &usize and there's no PartialEq impl for that

Reply:
Ok!